File: w12/d02/Classwork/minutes.js
function ourParseTime(time){
  var hour = parseInt( time.split(':')[0] );
  var minutes = parseInt( time.split(':')[1] );
  var meridian = time.slice(time.length - 2);

  return { hours: hour, minutes: minutes, meridian: meridian }
};

ourParseTime("12:00pm");
// { hours: 12, minutes: 0, meridian: 'pm' }


function minutesFromMidnight(time) {
  var totalMinutes = (time['hours'] % 12 * 60) + time['minutes'];

  if (time['meridian'] == 'pm') {
    totalMinutes = totalMinutes + 720;
  }

  return totalMinutes;
}


minutesFromMidnight({ hours: 12, minutes: 0, meridian: 'am' })
// 0
minutesFromMidnight({ hours: 12, minutes: 0, meridian: 'pm' })
// 720


function countingMinutes(timeSpan) {
  startTime = ourParseTime( timeSpan.split('-')[0] );
  endTime = ourParseTime( timeSpan.split('-')[1] );

  var totalMinutes = minutesFromMidnight(endTime) - minutesFromMidnight(startTime);

  if (totalMinutes < 0) {
    totalMinutes = totalMinutes + 1440;
  }

  return totalMinutes;
};


console.log(countingMinutes("12:30pm-01:00pm"))
// 30

console.log(countingMinutes("01:30am-01:15am"))
// 1425

File: w12/d02/Classwork/minutes.rb
def same_part_of_day(start_hours, end_hours, start_minutes, end_minutes)
	if (end_hours.to_i - start_hours.to_i) < 0 
		total_hours = (end_hours.to_i - start_hours.to_i) + 12 
		total_minutes = total_hours * 60
	else 
		total_hours = end_hours.to_i - start_hours.to_i
		total_minutes = total_hours * 60
	end 
	return total_minutes += end_minutes.to_i - start_minutes.to_i 
end 

def different_part_of_day(start_hours, end_hours, start_minutes, end_minutes)
	if end_hours.to_i - start_hours.to_i != 0 
		total_hours = (end_hours.to_i - start_hours.to_i) + 12
		total_minutes = total_hours * 60
	else 
		total_hours = 12
		total_minutes = total_hours.to_i * total_minutes.to_i
	end 
	return total_minutes += end_minutes.to_i - start_minutes.to_i
end 

def counting_time(time)
	two_times_split = time.split("-")
	# [12:30pm, 1:30pm]
	start_time = two_times_split[0]
	end_time = two_times_split[1]
	
	start_hours = start_time.split(":")[0]
	# 12
	start_minutes_with_am_pm = start_time.split(":")[1]
	# 30pm
	start_minutes = start_minutes_with_am_pm[0..1]
	# 30
	
	end_hours = end_time.split(":")[0]
	# 1
	end_minutes_with_am_pm = end_time.split(":")[1]
	# 30pm
	end_minutes = end_minutes_with_am_pm[0..1]
	# 30


	if start_time.index("pm") != nil && end_time.index("pm") != nil
		puts total_minutes = same_part_of_day(start_hours, end_hours, start_minutes, end_minutes)
	end  
	if start_time.index("am") != nil && end_time.index("am") != nil
		puts total_minutes = same_part_of_day(start_hours, end_hours, start_minutes, end_minutes)
	end 
	if start_time.index("am") != nil && end_time.index("pm") != nil
		puts different_part_of_day(start_hours, end_hours, start_minutes, end_minutes)
	end 
	if start_time.index("pm") != nil && end_time.index("am") != nil
		puts different_part_of_day(start_hours, end_hours, start_minutes, end_minutes)
	end 
end 

counting_time("1:23am-1:08am")
counting_time("12:30pm-12:00am")



